### ISSUE TITLE
gnupg 2.2.18 segfault on decrypt

### ISSUE THREAD

**Issue by joemiller** [Homebrew/homebrew-core#47469]
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link? https://gist.github.com/1e2f489e8a2410c33b1b2fdee54aad6a
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

Upgraded to gnupg 2.2.18 using the bottle and it segfaults on decryption. Rebuilt from source and it still segfaults.

## What happened (include command output)

<details>
  <summary>Command output</summary>
  <pre>

$ gpg -e --armor | gpg -d
foo
^D

gpg: signal Segmentation fault caught ... exiting
[1]    2683 done                gpg -e --armor |
       2684 segmentation fault  gpg -d

  </pre>
</details>

## What you expected to happen

decrypt the data, don't segfault

## Step-by-step reproduction instructions (by running `brew install` commands)

```console
$ brew upgrade gnupg

$ gpg -e --armor | gpg -d
foo
^D

gpg: signal Segmentation fault caught ... exiting
[1]    2683 done                gpg -e --armor |
       2684 segmentation fault  gpg -d
```

**Comment by hmrbarros** [comment]
As a work around I reverted homebrew-core taps to the previous commit 

```
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula
git checkout 8a66449def3ad5a5715ccdb1bc24aeb0a3ea5fcb
brew uninstall --ignore-dependencies gnupg
HOMEBREW_NO_AUTO_UPDATE=1 brew install gnupg
```

**Comment by iMichka** [comment]
> Upgraded to gnupg 2.2.18 using the bottle and it segfaults on decryption. Rebuilt from source and it still segfaults.

This is a sign that this is probably a `gnupg` issue, not an homebrew issue. Please report this to the `gnupg` bug tracker.

**Comment by joemiller** [comment]
fixed in 2.2.19
